pdns/rcpgenerator.cc

```diff
@@ -377,30 +377,36 @@ void RecordTextReader::xfrText(string& val, bool multi, bool lenField)
       val.append(1, ' ');
 
     skipSpaces();
-    if(d_string[d_pos]!='"') { // special case 'plenus' - without quotes
-      string::size_type pos = d_pos;
-      while(pos != d_end && isalnum(d_string[pos]))
-        pos++;
-      if(pos == d_end) {
-        val.append(1, '"');
-        val.append(d_string.c_str() + d_pos, d_end - d_pos);
-        val.append(1, '"');
-        d_pos = d_end;
+    bool is_plenus = false;
+    if(d_string[d_pos]=='"')
+      d_pos++;
+    else    // special case 'plenus' - without quotes
+      is_plenus = true;
+
+    val.append(1, '"');
+    for(; d_pos < d_end; d_pos++) {
+      if(d_string[d_pos]=='"') {
+        if(is_plenus)
+          throw RecordTextException("Unquoted data field in DNS should not contain a quote (\") in '"+d_string+"'");
+
+        // Otherwise we reached the end quote
         break;
       }
-      throw RecordTextException("Data field in DNS should start with quote (\") at position "+std::to_string(d_pos)+" of '"+d_string+"'");
-    }
-    val.append(1, '"');
-    while(++d_pos < d_end && d_string[d_pos]!='"') {
-      if(d_string[d_pos]=='\\' && d_pos+1!=d_end) {
+
+      if(is_plenus && dns_isspace(d_string[d_pos]))
+        break;
+
+      if(d_string[d_pos]=='\\' && d_pos+1!=d_end)
         val.append(1, d_string[d_pos++]);
```

Comment:
This code fails to reject the following invalid cases (but I don't know if they get caught elsewhere):
* too-short \\DDD escapes (e.g., `foo\03bar`)
* out-of-range \\DDD escapes (e.g., `foo\256bar`)
* unterminated escape sequences (e.g., `foo\`)

Reply:
Is this what's holding up the patch @gibson042

Reply:
They should certainly be fixed in my opinion, but I'm not a maintainer of this repository.